shoop/xtheme/plugins/widgets.py

```diff
@@ -90,3 +91,38 @@ def decompress(self, value):
         else:
             out_dict[FALLBACK_LANGUAGE_CODE] = value
         return out_dict
+
+
+class XThemeModelChoiceWidget(forms.Select):
+    def render(self, name, value, attrs=None, choices=()):
+        return mark_safe(
+            render_to_string("shoop/xtheme/_model_widget.jinja", {
+                "name": name,
+                "selected_object": [object for pk, object in self.choices if pk == int(value)][0] if value else None,
```

Comment:
If there are 15,000 products, this will iterate over all of them, then pick one. (I believe we have a helper called `first()` somewhere...)

Also, there's nothing guaranteeing pks are integers; comparing as strings would be safer.

It would really be better, all in all, that the ChoiceField passed the selected object itself down to the widget for rendering.